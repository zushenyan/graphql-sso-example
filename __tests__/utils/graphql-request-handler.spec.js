const graphqlRequestHandler = require("utils/graphql-request-handler.js");

describe("", () => {
  describe("request handler", () => {
    it("should work", async () => {
      const num = 42;
      const fn = async () => await Promise.resolve(num);
      await expect(graphqlRequestHandler(fn)).resolves.toMatchSnapshot();
    });

    it("should throw", async () => {
      const fn = async () => await Promise.reject("Q_Q");
      await expect(graphqlRequestHandler(fn)).resolves.toMatchSnapshot();
    });
  });
});
