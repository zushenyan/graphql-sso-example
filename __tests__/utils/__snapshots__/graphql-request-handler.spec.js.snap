// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` request handler should throw 1`] = `
Object {
  "message": "internal server error",
  "status": 500,
}
`;

exports[` request handler should work 1`] = `42`;
